### PR TITLE
Update EIP-2135: Move to Review

### PR DESCRIPTION
**Pull request by xinbenlv**
When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: EIPS/eip-2135.md [hunk 1]
```diff
@@ -1,80 +1,103 @@
 ---
 eip: 2135
-title: Consumable Interface
+title: NFT Consumables
```
Comment (SamWilsn) [inline EIPS/eip-2135.md:3]: Bikeshedding the title a bit. The title should include a bit more context about what "consume" means.

```suggestion
title: One Time Use NFTs
```

```suggestion
title: Limited Use NFTs
```

File: EIPS/eip-2135.md [hunk 2]
```diff
@@ -1,80 +1,103 @@
...
+description: An interface extending EIP-721 and EIP-1155 for consumability.
```
Comment (SamWilsn) [inline EIPS/eip-2135.md:4]:
```suggestion
description: An interface extending EIP-721 and EIP-1155 with ticket-like functionality.
```

File: EIPS/eip-2135.md [hunk 3]
```diff
@@ -1,80 +1,103 @@
...
+  /// @param _consumer the address of consumer of this token. It doesn't have
+  ///                  to be the EOA or contract Account that initiates the TX.
+  /// @param _assetId  the NFT asset being consumed
+  /// @param _data     extra data passed in for consume for extra message
+  ///                  or future extension.
+  event OnConsumption(address indexed _consumer, uint256 indexed _assetId, bytes[] _data);
 }
 ```
 
+2. If the compliant contract is an [EIP-721](./eip-721.md) or [EIP-1155](./eip-1155.md), in addition to `OnConsumption`, it **MUST** also emit related `Transfer` / `TransferSingle` event as if a token has been transfer from the current holder to `0x00` address
```
Comment (SamWilsn) [inline EIPS/eip-2135.md:59]: Is it required to implement one of EIP-721 or EIP-1155, or is that optional?

File: EIPS/eip-2135.md [hunk 4]
```diff
@@ -1,80 +1,103 @@
...
+  /// @param _assetId  the NFT asset being consumed
+  function isConsumableBy(address _consumer, uint256 _assetId) external view returns (bool _consumable);
+
+  /// @notice The event emitted when there is a successful consumption.
+  /// @param _consumer the address of consumer of this token. It doesn't have
+  ///                  to be the EOA or contract Account that initiates the TX.
+  /// @param _assetId  the NFT asset being consumed
+  /// @param _data     extra data passed in for consume for extra message
+  ///                  or future extension.
+  event OnConsumption(address indexed _consumer, uint256 indexed _assetId, bytes[] _data);
```
Comment (SamWilsn) [inline EIPS/eip-2135.md:55]: Maybe `Consume`? That would match how `Transfer` is named in EIP-721.

File: EIPS/eip-2135.md [hunk 5]
```diff
@@ -1,80 +1,103 @@
...
 
- - who has the power to perform such activity. 
- - under what condition such consumption can occur.
- 
-It does, however, assume the asset can be identified in a `uint256` assetId as in th parameter. A design convention and compatibility consideration is put in place to follow the ERC-721
+- who has the power to perform such activity.
+- under what condition such consumption can occur.
 
-The event notifies subscribers whoever are interested to learn an asset is being consumed. The boolean function of `isConsumable` can be used to check whether an asset is still consumable. 
+It does, however, assume the asset can be identified in a `uint256` assetId as in th parameter. A design convention and compatibility consideration is put in place to follow the EIP-721
```
Comment (SamWilsn) [inline EIPS/eip-2135.md:74]:
```suggestion
It does, however, assume the asset can be identified in a `uint256` asset id as in the parameter. A design convention and compatibility consideration is put in place to follow the EIP-721
```

File: EIPS/eip-2135.md [hunk 6]
```diff
@@ -1,80 +1,103 @@
...
 
-To keep it simple, this standard *intentionally* contains no functions or events related to creation of a consumable asset. This is because the creation of a consumable asset will need to make assumption of the nature of an actual use-case. If we see some common use-case of creation, we can have another follow up standard.
+2. The event notifies subscribers whoever are interested to learn an asset is being consumed. The boolean function of `isConsumable` can be used to check whether an asset is still consumable.
```
Comment (SamWilsn) [inline EIPS/eip-2135.md:76]: If the asset is transferred to the zero address, is it ever consumable again?

File: EIPS/eip-2135.md [hunk 7]
```diff
@@ -1,80 +1,103 @@
...
 
-We also left out metadata associated to the consumables from the standard. If necessary, related metadata can be created with a separate metadata extension interface like [`ERC-721 Metadata`](./eip-721.md)
+3. To keep it simple, this standard *intentionally* contains no functions or events related to creation of a consumable asset. This is because the creation of a consumable asset will need to make assumption of the nature of an actual use-case. If we see some common use-case of creation, we can have another follow up standard.
 
-## Backwards Compatibility
+4. We also left out metadata associated to the consumables from the standard. If necessary, related metadata can be created with a separate metadata extension interface like [`EIP-721 Metadata`](./eip-721.md)
```
Comment (SamWilsn) [inline EIPS/eip-2135.md:80]:
```suggestion
4. We also left out metadata associated to the consumables from the standard. If necessary, related metadata can be created with a separate metadata extension interface like `ERC721Metadata` from [EIP-721](./eip-721.md)
```

File: EIPS/eip-2135.md (hunk 1 above)
Comment (xinbenlv) [reply EIPS/eip-2135.md:3]: Thanks for suggestion. Finding a good name is hard! 
For now I'd update the name `Consumable NFT for Ticketing`. Will look forward to community to give feedback

File: EIPS/eip-2135.md (hunk 2 above)
Comment (xinbenlv) [reply EIPS/eip-2135.md:4]: Updated slightly differently. Will recommit

File: EIPS/eip-2135.md (hunk 4 above)
Comment (xinbenlv) [reply EIPS/eip-2135.md:55]: I feel differentiating the event vs function just by capitalization is a little less clear. (From my language background English-as-second-language)

File: EIPS/eip-2135.md (hunk 3 above)
Comment (xinbenlv) [reply EIPS/eip-2135.md:59]: Good question! I added an answer in rationale section

File: EIPS/eip-2135.md (hunk 6 above)
Comment (xinbenlv) [reply EIPS/eip-2135.md:76]: No, it will not be consumable again. You made a good point. Let me make it more clear.

**Comment by Pandapip1** [comment]
@xinbenlv mind reviewing my changes?

**Comment by xinbenlv** [comment]
LGTM, thank you for the reviews @Pandapip1 . All look good

**Comment by xinbenlv** [comment]
@Pandapip1 do you have any other comments on this PR?

**Review by Pandapip1 — APPROVED** [review]
LGTM

**Pull request merged**